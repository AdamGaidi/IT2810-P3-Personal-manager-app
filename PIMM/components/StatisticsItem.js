  import React, { Component } from "react";
import { StyleSheet, Text, View } from "react-native";
import Colors from "../constants/Colors";
import Strings from "../constants/Strings";

export default class StatisticsItem extends Component {
  render() {
    const { text, steps } = this.props;
    return (
      <View style={styles.staticsItemWrapper}>
        <Text style={styles.statsText}>{text.toUpperCase()}</Text>
        <View style={styles.statsStepWrapper}>
          <Text style={styles.statsStep}>{steps} steps</Text>
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  staticsItemWrapper: {
    flex: 1,
    flexDirection: "row",
    justifyContent: "space-between",
    marginVertical: 4
  },
  statsText: {
    color: Colors.primaryBlue,
    fontSize: 15
  },
  statsStepWrapper: {
    borderBottomColor: Colors.primaryBlue,
    borderBottomWidth: 1.5
  },
  statsStep: {
    color: Colors.taskGray,
    textAlign: "right",
    fontSize: 15
  }
});
