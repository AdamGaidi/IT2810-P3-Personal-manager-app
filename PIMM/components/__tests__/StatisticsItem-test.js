import React from "react";
import StatisticsItem from "../StatisticsItem";
import renderer from "react-test-renderer";

test("renders correctly with example props", () => {
  const tree = renderer
    .create(<StatisticsItem text={"eksempel"} steps={0} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
