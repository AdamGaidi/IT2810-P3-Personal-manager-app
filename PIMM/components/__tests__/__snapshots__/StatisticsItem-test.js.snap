// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with example props 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginVertical": 4,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#5a95ff",
        "fontSize": 15,
      }
    }
  >
    EKSEMPEL
  </Text>
  <View
    style={
      Object {
        "borderBottomColor": "#5a95ff",
        "borderBottomWidth": 1.5,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#545454",
          "fontSize": 15,
          "textAlign": "right",
        }
      }
    >
      0
       steps
    </Text>
  </View>
</View>
`;
